src/AssetsController.ts

```diff
@@ -651,13 +672,20 @@ export class AssetsController extends BaseController<AssetsConfig, AssetsState>
 	 */
 	removeToken(address: string) {
 		address = toChecksumAddress(address);
-		const { allTokens, tokens } = this.state;
+		const { allTokens, tokens, ignoredTokens } = this.state;
 		const { networkType, selectedAddress } = this.config;
-		const newTokens = tokens.filter((token) => token.address !== address);
+		const newIgnoredTokens = [...ignoredTokens];
+		const newTokens = tokens.filter((token) => {
+			/* istanbul ignore else */
+			if (token.address === address) {
+				newIgnoredTokens.push(token);
```

Comment:
I believe here (and doing the same for collectibles) would duplicate `ignoredTokens` if he user decides to remove a token twice or more (edge case), there is no validation that a token is already on that list.

Reply:
Added check to prevent dupes and added tests for that scenario